src/ATokenFlexVoting.sol

```diff
@@ -214,32 +187,35 @@ contract ATokenFlexVoting is AToken {
     }
   }
 
-  /// @notice Causes this contract to cast a vote to the Governor for all the
-  /// tokens it currently holds. Uses the sum of all depositor voting
-  /// expressions to decide how to split its voting weight. Can be called by
-  /// anyone, but _must_ be called within `CAST_VOTE_WINDOW` blocks before the
-  /// proposal deadline. We don't bother to check if the vote has already been
-  /// cast -- GovernorCountingFractional will revert if it has.
+  /// @notice Causes this contract to cast a vote to the Governor for all of the
+  /// accumulated votes expressed by users. Uses the sum of all raw (unrebased) balances
+  /// to proportion and split its voting weight. Can be called by anyone. Can be called
+  /// multiple times during the lifecycle of a given proposal.
   /// @param proposalId The ID of the proposal which the Pool will now vote on.
   function castVote(uint256 proposalId) external {
-    require(
-      internalVotingPeriodEnd(proposalId) <= block.number,
-      "cannot castVote during internal voting period"
-    );
-
     ProposalVote storage _proposalVote = proposalVotes[proposalId];
     require(
       _proposalVote.forVotes + _proposalVote.againstVotes + _proposalVote.abstainVotes > 0,
       "no votes expressed"
     );
-
     uint256 _proposalSnapshotBlockNumber = GOVERNOR.proposalSnapshot(proposalId);
 
-    // Use the snapshot of total raw balances to determine total voting weight.
-    // We cannot use the proposalVote numbers alone, since some people with
-    // balances at the snapshot might not have expressed votes. We don't want to
-    // make it possible for aToken holders to *increase* their voting power when
-    // other people don't express their votes. That'd be a terrible incentive.
+    // We use the snapshot of total raw balances to determine the weight with
+    // which to vote. We do this for two reasons:
+    //   (1) We cannot use the proposalVote numbers alone, since some people with
+    //       balances at the snapshot might never express their preferences. If a
+    //       large holder never expressed a preference, but this contract nevertheless
+    //       cast votes to the governor with all of its weight, then other users would
```

Comment:
Small clarification since it's not guaranteed that your voting weight would have increased, since it's also dependent on how many tokens are borrowed (which isn't mentioned in this comment block, but maybe should be)
```suggestion
    //       cast votes to the governor with all of its weight, then other users may
```